lib/puppet/http/service/ca.rb

```diff
@@ -1,12 +1,42 @@
+#
+# @api private
+#
+# The Ca service is used to handle certificate requests
+#
 class Puppet::HTTP::Service::Ca < Puppet::HTTP::Service
+  # @api private
+  # @return [Hash] default headers for the ca service
   HEADERS = { 'Accept' => 'text/plain' }.freeze
+
+  # @api private
+  # @return [String] default API for the ca service
   API = '/puppet-ca/v1'.freeze
 
+  #
+  # @api private
+  #
+  # @param [Puppet::HTTP::Client] client
+  # @param [Puppet::HTTP::Session] session
+  # @param [String] server (Puppet[:ca_server]) The default server used if the
+  #   settings resolver is used
+  # @param [Integer] port (Puppet[:ca_port])  The default port used if the
+  #   settings resolver is used
```

Comment:
Ah the word `default` is tripping me up. If the `server` and `port` arguments are given, then the caller wants to create a service for that explicit server/port. If the arguments are `nil` it means use the default values from settings for that kind of service.